src/common/http/http.cpp

```diff
@@ -188,6 +188,23 @@ string URLEncode(const string &value) {
 	return escaped.str();
 }
 
+string URLDecode(const string &value) {
+	stringstream unescaped;
+
+	auto len = value.length();
+	for (size_t i = 0; i < len; i++) {
+		if (value[i] != '%') {
+			unescaped << value[i];
+		} else {
+			unsigned int num;
+			sscanf(value.substr(i + 1, 2).c_str(), "%x", &num);
```

Comment:
Should we add a boundary check here (if `%` is one of the last 2 characters)?

Reply:
Hmm, actually, the function should return `ExpectedString` and error in this case (and others).